File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import NotificationKind


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    kind: NotificationKind
    title: str
    body: str | None
    read_at: datetime | None
    created_at: datetime

File: frontend/src/types/notification.ts
export type NotificationKind = "reminder" | "daily_digest" | "unfinished";

export type AppNotification = {
  id: number;
  kind: NotificationKind;
  title: string;
  body: string | null;
  read_at: string | null;
  created_at: string;
};

export const NOTIFICATION_LABEL: Record<NotificationKind, string> = {
  reminder: "リマインダー",
  daily_digest: "今日のまとめ",
  unfinished: "やり残し",
};
